canonical/ubuntu-advantage-client
# docs: separate user docs from developer docs

## Notes

The intent of this PR is to just begin the process of organizing some of our existing documentation.

I've separated out what I think are more developer-oriented docs and moved them to a new `CONTRIBUTING.md` file.

That means the `README.md` will be more of a user-facing document.

I also started a `contributing-docs` folder to hold individual pieces of content, separate from the soon-to-exist `docs` folder which will hold user-facing documentation.

This PR only involves moving existing content around - not actual content changes (except for some small corrections and a small attempt to make our readme look more fun).

I expect next steps will be breaking individual pieces of these large documents into their own files and tailoring them to a particular documentation type (e.g. howtoguide, explanation, reference), one at a time.

## Proposed Commit Message


```
docs: separate user docs from developer docs
```

## Test Steps

Read the docs. 

Make sure I didn't drop any content by mistake.

Make sure the separation makes some sense.

## Desired commit type


## Checklist
